pkgs/applications/system/howdy/default.nix

```diff
@@ -0,0 +1,152 @@
+{ stdenv
+, lib
+, fetchFromGitHub
+, fetchurl
+, bzip2
+, gobject-introspection
+, meson
+, ninja
+, pkg-config
+, wrapGAppsHook
+, makeDesktopItem
+, makeWrapper
+, fmt
+, gettext
+, gtk3
+, inih
+, libevdev
+, pam
+, python3
+}:
+
+let
+  data =
+    let
+      baseurl = "https://github.com/davisking/dlib-models/raw/daf943f7819a3dda8aec4276754ef918dc26491f";
+    in
+    {
+      "dlib_face_recognition_resnet_model_v1.dat" = fetchurl {
+        url = "${baseurl}/dlib_face_recognition_resnet_model_v1.dat.bz2";
+        sha256 = "0fjm265l1fz5zdzx5n5yphl0v0vfajyw50ffamc4cd74848gdcdb";
+      };
+      "mmod_human_face_detector.dat" = fetchurl {
+        url = "${baseurl}/mmod_human_face_detector.dat.bz2";
+        sha256 = "117wv582nsn585am2n9mg5q830qnn8skjr1yxgaiihcjy109x7nv";
+      };
+      "shape_predictor_5_face_landmarks.dat" = fetchurl {
+        url = "${baseurl}/shape_predictor_5_face_landmarks.dat.bz2";
+        sha256 = "0wm4bbwnja7ik7r28pv00qrl3i1h6811zkgnjfvzv7jwpyz7ny3f";
+      };
+    };
+      # wrap howdy and howdy-gtk
+
+  py = python3.withPackages (p: [
+    p.elevate
+    p.face-recognition
+    p.keyboard
+    (p.opencv4.override { enableGtk3 = true; })
+    p.pycairo
+    p.pygobject3
+  ]);
+
+  desktopItem = makeDesktopItem {
+    name = "howdy";
+    exec = "howdy-gtk";
+    icon = "howdy";
+    comment = "Howdy facial authentication";
+    desktopName = "Howdy";
+    genericName = "Facial authentication";
+    categories = [ "System" "Security" ];
+  };
+in
+stdenv.mkDerivation {
+  pname = "howdy";
+  version = "unstable-2023-09-14";
+
+  src = fetchFromGitHub {
+    owner = "boltgolt";
+    repo = "howdy";
+    rev = "95df4d2d48d772d88ad1a8f2be290ce38752bdd4";
```

Comment:
Maybe updating the git hash would fix it? There has been new updates to howdy since this hash version (which is from September)

Reply:
Reference: https://github.com/boltgolt/howdy/compare/95df4d2d48d772d88ad1a8f2be290ce38752bdd4...beta

Reply:
Seems to work after rebuilding and rebooting, even after suspending/resuming. But it's too early to tell whether it's completely fixed.

Reply:
It looks like for a bit it worked, but then it proceeded to crash on me again and completely lock up. 🤔 Could it be something python related?

Reply:
Okay, it broke for me again. I'll have to investigate what's going on.

Reply:
Really odd, I haven't had it break on me again

Reply:
I actually had it crash again today randomly.. I'm really unsure why exactly it's doing that, I think it seems to happen after there's a failed attempt? 🤔

Reply:
In my case it will happen after suspending/resuming the laptop. I currently work around it by typing my password and then pressing enter, while not being in the line of sight of the camera (or covering it).

Reply:
I managed to replicate the crash the exact same way! I let my laptop sleep with the screen open and then woke it up and that's when howdy locks up. It looks like something with suspending and resuming that causes howdy to break.